locals.tf

```diff
@@ -117,4 +117,7 @@ locals {
       vnet_subnet_id = var.subnet_id
       max_pods       = var.max_pods_per_node
   } } : null)
+
+  log_analytics_workspace_name                     = coalesce(var.log_analytics_workspace_name_override, "${var.name}-log-analytics")
+  log_analytics_cluster_autoscaler_diagnostic_name = coalesce(var.cluster_autoscaler_diagnostic_name_override, "${var.name}-cluster-autoscaler-diagnostic")
```

Comment:
please ensure this 
`The name should include 4-63 letters, digits or '-'. The '-' shouldn't be the first or the last symbol`